Nightly ITs: `bigquery` IT fails due to quota limit exceeded

From https://github.com/GoogleCloudPlatform/spring-cloud-gcp/actions/runs/8229523346/job/22500895100

```
Error: 1:230 [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 20.18 s <<< FAILURE! -- in com.google.cloud.spring.bigquery.integration.outbound.BigQueryFileMessageHandlerIntegrationTests
Error: 1:231 [ERROR] com.google.cloud.spring.bigquery.integration.outbound.BigQueryFileMessageHandlerIntegrationTests.testLoadFile_cancel -- Time elapsed: 4.512 s <<< ERROR!
com.google.cloud.bigquery.BigQueryException: Exceeded rate limits: too many table update operations for this table. For more information, see https://cloud.google.com/bigquery/docs/troubleshoot-quotas
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:115)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.deleteTable(HttpBigQueryRpc.java:381)
	at com.google.cloud.bigquery.BigQueryImpl$10.call(BigQueryImpl.java:593)
	at com.google.cloud.bigquery.BigQueryImpl$10.call(BigQueryImpl.java:590)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.bigquery.BigQueryImpl.delete(BigQueryImpl.java:589)
	at com.google.cloud.spring.bigquery.integration.outbound.BigQueryFileMessageHandlerIntegrationTests.setup(BigQueryFileMessageHandlerIntegrationTests.java:74)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
```

--- a/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
+++ b/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
@@ -76,7 +76,9 @@ class BigQueryFileMessageHandlerIntegrationTests {
   @BeforeEach
   @AfterEach
   void setup() {
-    tableName = TABLE_NAME_PREFIX + UUID.randomUUID();
+    if (tableName == null) {
+      tableName = TABLE_NAME_PREFIX + UUID.randomUUID();
+    }
     // Clear the previous dataset before beginning the test.
     this.bigquery.delete(TableId.of(DATASET_NAME, tableName));
   }

--- a/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
+++ b/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
@@ -60,8 +60,6 @@ import org.springframework.test.context.junit.jupiter.SpringExtension;
 @DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
 class BigQueryFileMessageHandlerIntegrationTests {
 
-  private static final String TABLE_NAME_PREFIX = "test_table_";
-
   private String tableName;
 
   @Autowired
@@ -77,7 +75,7 @@ class BigQueryFileMessageHandlerIntegrationTests {
   @AfterEach
   void setup() {
     if (tableName == null) {
-      tableName = TABLE_NAME_PREFIX + UUID.randomUUID().toString().replace('-', '_');
+      tableName = "test_table_" + UUID.randomUUID().toString().replace('-', '_');
     }
     // Clear the previous dataset before beginning the test.
     this.bigquery.delete(TableId.of(DATASET_NAME, tableName));

--- a/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
+++ b/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
@@ -77,7 +77,7 @@ class BigQueryFileMessageHandlerIntegrationTests {
   @AfterEach
   void setup() {
     if (tableName == null) {
-      tableName = TABLE_NAME_PREFIX + UUID.randomUUID();
+      tableName = TABLE_NAME_PREFIX + UUID.randomUUID().toString().replace('-', '_');
     }
     // Clear the previous dataset before beginning the test.
     this.bigquery.delete(TableId.of(DATASET_NAME, tableName));
@@ -113,7 +113,8 @@ class BigQueryFileMessageHandlerIntegrationTests {
     jobFuture.get();
 
     QueryJobConfiguration queryJobConfiguration =
-        QueryJobConfiguration.newBuilder("SELECT * FROM test_dataset.test_table").build();
+        QueryJobConfiguration.newBuilder(String.format("SELECT * FROM test_dataset.%s", tableName))
+            .build();
     TableResult result = this.bigquery.query(queryJobConfiguration);
 
     assertThat(result.getTotalRows()).isEqualTo(1);
@@ -148,7 +149,8 @@ class BigQueryFileMessageHandlerIntegrationTests {
     jobFuture.get();
 
     QueryJobConfiguration queryJobConfiguration =
-        QueryJobConfiguration.newBuilder("SELECT * FROM test_dataset.test_table").build();
+        QueryJobConfiguration.newBuilder(String.format("SELECT * FROM test_dataset.%s", tableName))
+            .build();
     TableResult result = this.bigquery.query(queryJobConfiguration);
 
     assertThat(result.getTotalRows()).isEqualTo(1);
@@ -175,7 +177,8 @@ class BigQueryFileMessageHandlerIntegrationTests {
     assertThat(job).isNotNull();
 
     QueryJobConfiguration queryJobConfiguration =
-        QueryJobConfiguration.newBuilder("SELECT * FROM test_dataset.test_table").build();
+        QueryJobConfiguration.newBuilder(String.format("SELECT * FROM test_dataset.%s", tableName))
+            .build();
     TableResult result = this.bigquery.query(queryJobConfiguration);
     assertThat(result.getTotalRows()).isEqualTo(1);
   }

--- a/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
+++ b/spring-cloud-gcp-bigquery/src/test/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandlerIntegrationTests.java
@@ -35,6 +35,7 @@ import com.google.cloud.spring.bigquery.integration.BigQuerySpringMessageHeaders
 import java.io.File;
 import java.time.Duration;
 import java.util.HashMap;
+import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
@@ -59,19 +60,25 @@ import org.springframework.test.context.junit.jupiter.SpringExtension;
 @DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
 class BigQueryFileMessageHandlerIntegrationTests {
 
-  private static final String TABLE_NAME = "test_table";
+  private static final String TABLE_NAME_PREFIX = "test_table_";
 
-  @Autowired private ThreadPoolTaskScheduler taskScheduler;
+  private String tableName;
 
-  @Autowired private BigQuery bigquery;
+  @Autowired
+  private ThreadPoolTaskScheduler taskScheduler;
 
-  @Autowired private BigQueryFileMessageHandler messageHandler;
+  @Autowired
+  private BigQuery bigquery;
+
+  @Autowired
+  private BigQueryFileMessageHandler messageHandler;
 
   @BeforeEach
   @AfterEach
   void setup() {
+    tableName = TABLE_NAME_PREFIX + UUID.randomUUID();
     // Clear the previous dataset before beginning the test.
-    this.bigquery.delete(TableId.of(DATASET_NAME, TABLE_NAME));
+    this.bigquery.delete(TableId.of(DATASET_NAME, tableName));
   }
 
   @Test
@@ -83,7 +90,7 @@ class BigQueryFileMessageHandlerIntegrationTests {
             Field.newBuilder("County", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build());
 
     HashMap<String, Object> messageHeaders = new HashMap<>();
-    messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, TABLE_NAME);
+    messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, tableName);
     messageHeaders.put(BigQuerySpringMessageHeaders.FORMAT_OPTIONS, FormatOptions.csv());
     messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_SCHEMA, schema);
 
@@ -119,7 +126,7 @@ class BigQueryFileMessageHandlerIntegrationTests {
   @Test
   void testLoadFile() throws InterruptedException, ExecutionException {
     HashMap<String, Object> messageHeaders = new HashMap<>();
-    this.messageHandler.setTableName(TABLE_NAME);
+    this.messageHandler.setTableName(tableName);
     this.messageHandler.setFormatOptions(FormatOptions.csv());
 
     Message<File> message =
@@ -155,7 +162,7 @@ class BigQueryFileMessageHandlerIntegrationTests {
     this.messageHandler.setSync(true);
 
     HashMap<String, Object> messageHeaders = new HashMap<>();
-    messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, TABLE_NAME);
+    messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, tableName);
     messageHeaders.put(BigQuerySpringMessageHeaders.FORMAT_OPTIONS, FormatOptions.csv());
 
     Message<File> message =
@@ -174,7 +181,7 @@ class BigQueryFileMessageHandlerIntegrationTests {
   @Test
   void testLoadFile_cancel() {
     HashMap<String, Object> messageHeaders = new HashMap<>();
-    messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, TABLE_NAME);
+    messageHeaders.put(BigQuerySpringMessageHeaders.TABLE_NAME, tableName);
     messageHeaders.put(BigQuerySpringMessageHeaders.FORMAT_OPTIONS, FormatOptions.csv());
 
     Message<File> message =

